front_end_testing.md

```diff
@@ -0,0 +1,32 @@
+---
+layout: default
+title: Front-end testing
+---
+
+# Front-end testing
+
+Sometimes we want to do UI (User Interface) testing. One thing we can use for this is [DalekJS](http://dalekjs.com/). We'll use it to test that the right pages are there, and that they work as expected.
+
+## Set up DalekJS
+
+DalekJS is available as an npm package, which makes installation convenient for us. Follow [the installation instructions](http://dalekjs.com/pages/getStarted.html#install).
+
+Install the dalek-cli globally ([like we did for the Mocha cli](http://node.projectcodex.co/node-tutorial/index.html)).
+
+```
+npm install dalek-cli -g
```

Comment:
@SteveBarnett this global install will need  to be done a super user on ubuntu `sudo` command so will need a `sudo`.